nixos/modules/services/web-apps/mastodon.nix

```diff
@@ -511,7 +554,16 @@ in {
     };
 
     systemd.services.mastodon-init-db = lib.mkIf cfg.automaticMigrations {
-      script = ''
+      script = lib.optionalString (!databaseActuallyCreateLocally) ''
```

Comment:
I really don't like the additional shell code here. Can't we replace this with some additions to systemd.services.mastodon-init-db.environment?

Reply:
I haven't found any other variants. And storing the database password in a variable is not secure.

Reply:
I also looked into it and it seems like there is no way to just provide a file with a password in it to postgresql. Only supported way seems to be the pgpassfile syntax. Let's leave it like this then :+1: